### PR TITLE
Add <10C filter,sss_smos_esa.yaml

### LINKED ISSUE #355: Passivate <10C in sss_smap_jpl.yaml
The following filter to passivate high-lat sss obs is not present in sss_smap_jpl.yaml.

Please ignore and close the issue if this is intended.

```
  - filter: Domain Check
    action:
      name: passivate
    where:
    - variable: {name: sea_surface_temperature@GeoVaLs}
      minvalue: 10.0
```

### LINKED ISSUE #357: Add sss_smos_esa.yaml file
The yaml file for SMOS ESA L2 SSS data is missing.

### PR DESCRIPTION
**Pull request by ShastriPaturi**
## Description
This PR adds the <10C "passivate" filter in sss_smap_jpl.yaml & sss_trak_fnmoc.yaml.
Creates sss_smos_esa.yaml that was missing.

### Issue(s) addressed
Closes #355 
Closes #357

### CODE REVIEW

**Comment by mjhossen** [comment]
The soca directory should be under gdas directory. Also, please remove comments, such as
```
  # Passivate obs where ocean fraction is > 90%
  #  - filter: Domain Check
  #    action:
  #      name: passivate
  #    where:
  #    - variable: {name: sea_area_fraction@GeoVaLs}
  #      maxvalue: 0.9
```
@mjhossen I am not sure I understand your comment "The soca directory should be under gdas directory". The soca directory is under godas. Also, please add these comments in the "Files changed" tab and start your review by clicking on "Review changes/Request changes"

File: soca/obs/sss_smap_jpl.yaml [hunk 1]
```diff
@@ -22,9 +22,19 @@
...
-  ## Gaussian_Thinning is having problems with LETKF, try again later
-  # - filter: Gaussian_Thinning
-  #   horizontal_mesh:   25.0 #km
+  - filter: Domain Check
+    action:
+      name: passivate
+    where:
+    - variable: {name: sea_surface_temperature@GeoVaLs}
+      minvalue: 10.0
+  # Passivate obs where ocean fraction is > 90%
```
Comment (guillaumevernieres) [inline soca/obs/sss_smap_jpl.yaml:31]: they should be passive by default , un-comment the lines below.

File: soca/obs/sss_smos_esa.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    maxvalue: 40.0
+  - filter: Background Check
+    threshold: 5.0
+  - filter: Domain Check
+    action:
+      name: passivate
+    where:
+    - variable: {name: sea_surface_temperature@GeoVaLs}
+      minvalue: 10.0
+  # Passivate obs where ocean fraction is > 90%
```
Comment (guillaumevernieres) [inline soca/obs/sss_smos_esa.yaml:31]: Same comment as above. sss is passive by default, at least for now.

File: soca/obs/sss_trak_fnmoc.yaml [hunk 3]
```diff
@@ -30,6 +30,13 @@
     where:
     - variable: {name: sea_surface_temperature@GeoVaLs}
       minvalue: 10.0
+  # Passivate obs where ocean fraction is > 90%
```
Comment (guillaumevernieres) [inline soca/obs/sss_trak_fnmoc.yaml:33]: ... Oops ... Looks like I was not paying attention, these should have been passive for sure. We need to be a bit more careful when we put these yaml files together. Ouch.

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]

**Review by guillaumevernieres — APPROVED** [review]

**Comment by liuxiao37k** [comment]
Do we never have a coastal grid with sea_area_fraction@GeoVaLs < 0.9? Just curious why a threshold of 0.9, not something like 0.999... I am asking because in L4 adt the passivation mechanism also includes a rejection for coastal obs with  sea_area_fraction@GeoVaLs < 0.9, which I understand (i.e. "never trust nearshore obs, even don't use them for validation")
```
# Reject obs where ocean fraction is < 90%
  - filter: Domain Check
    action:
      name: reject
    where:
    - variable: {name: sea_area_fraction@GeoVaLs}
      minvalue: 0.9
  # Passivate obs where ocean fraction is > 90%
  - filter: Domain Check
    action:
      name: passivate
    where:
    - variable: {name: sea_area_fraction@GeoVaLs}
      maxvalue: 0.9
```

**Comment by guillaumevernieres** [comment]
> Do we never have a coastal grid with sea_area_fraction@GeoVaLs < 0.9? Just curious why a threshold of 0.9, not something like 0.999... I am asking because in L4 adt the passivation mechanism also includes a rejection for coastal obs with sea_area_fraction@GeoVaLs < 0.9, which I understand (i.e. "never trust nearshore obs, even don't use them for validation")
> 
> ```
> # Reject obs where ocean fraction is < 90%
>   - filter: Domain Check
>     action:
>       name: reject
>     where:
>     - variable: {name: sea_area_fraction@GeoVaLs}
>       minvalue: 0.9
>   # Passivate obs where ocean fraction is > 90%
>   - filter: Domain Check
>     action:
>       name: passivate
>     where:
>     - variable: {name: sea_area_fraction@GeoVaLs}
>       maxvalue: 0.9
> ```

@liuxiao37k , these filters work but definitely confuse me, even though I probably was the one originally adding that stuff to the `yaml`. 
The mask is 0 or 1, nothing in between. Happy to revisit the filters with something that makes more sense, but probably not in this PR.

**Review by liuxiao37k — APPROVED** [review]

**Comment by liuxiao37k** [comment]
> @liuxiao37k , these filters work but definitely confuse me, even though I probably was the one originally adding that stuff to the `yaml`. The mask is 0 or 1, nothing in between. Happy to revisit the filters with something that makes more sense, but probably not in this PR.

@guillaumevernieres Thanks! Sounds good. I have been checking and I can see these filters work just fine, but curious if there is any history behind the number 0.9.

**Comment by guillaumevernieres** [comment]
> > @liuxiao37k , these filters work but definitely confuse me, even though I probably was the one originally adding that stuff to the `yaml`. The mask is 0 or 1, nothing in between. Happy to revisit the filters with something that makes more sense, but probably not in this PR.
> 
> @guillaumevernieres Thanks! Sounds good. I have been checking and I can see these filters work just fine, but curious if there is any history behind the number 0.9.

yes, it's an interesting story. @guillaumevernieres was having a senior moment and forgot that the mask was 0 OR 1 ... and got confused in the other instance.

**Pull request merged**